### DISCUSSION TITLE
How to provide communication between dialog component and footer in DynamicDialog

### DISCUSSION THREAD

**Discussion by mconner** [primefaces/community#1017] [PrimeNG] [answered]
By the DynamicDialog stackblitz example in the documentation, I see that there's a way to define a Footer Template as a component.  However, there's no apparent mechanism for defining how to communicate between the footer and the component used within the dialog content.  How would you:  
- Access the form defined in dialog's componentType template, for example, to see if it is valid (and disable a save button if it is).  
- Submit the form if the user clicks on the save button in the component.  

I could simply style part of the dialog's component to be a footer (p-dialog-footer), but then the scrolling on resize doesn't work properly.

EDIT: I've worked around this by using the DynamicDialogConfig.data, populating callbacks in the dialog componentType (which has the form as a ViewChild), for the footer to call to create SAVE/CANCEL buttons.

**Comment by cetincakiroglu** [comment]
Hi,

In this case, you can create your custom service logic to communicate between components. If you provide a stackblitz example, you can receive assistance more easily.

**Comment by mconner** [comment]
@cetincakiroglu : Yes, I thought of that as well (after posting the question) and may switch my workaround to that. However, this seems like basic functionality, and should be more directly supported in the component.

**Reply by cetincakiroglu** [reply, marked as answer]
We cannot guess what methods or functionality that users might use. There are a lot of `what if` scenarios. The best solution is to create your service and do as you wish.

**Comment by kashcode** [comment]
@mconner  Hi, can you share your solution to this problem currently I have same issue and trying figuring out how solve it.
